ibis/backends/pandas/tests/execution/conftest.py

```diff
@@ -62,9 +63,17 @@ def df():
             .dt.tz_localize('UTC')
             .astype(str),
             'decimal': list(map(decimal.Decimal, ['1.0', '2', '3.234'])),
-            'array_of_float64': [[1.0, 2.0], [3.0], []],
-            'array_of_int64': [[1, 2], [], [3]],
-            'array_of_strings': [['a', 'b'], [], ['c']],
+            'array_of_float64': [
+                np.array([1.0, 2.0]),
```

Comment:
are we allowed to have a None / np.nan? (e.g. a missing value for the entire row?)

Reply:
Yes, this should not lead to any issues, besides possibly if we're trying to infer the Ibis dtype of the array.

If it contains `np.nans` then it would be OK (inferred dtype would be `dt.Array(dt.float64)`).

If it contains `Nones` then the inferred dtype would be more general, e.g. `dt.Array(dt.binary)`.

Reply:
no what i mean is the array itself is None e.g. `[np.array(...), np.array(...), None]` (this also maybe restricted / not allowed). If you can followup with tests for this.

Reply:
This technically will work but I wouldn't consider it well-supported

Type inference can be imperfect in this case (sort of "best-effort"): To infer the type of an `object` Series (includes Series that have `np.arrays`, `Nones`, or a mix, etc), I check the type of the first element of the Series. So in your example, it depends on whether a `None` or a `np.array(...)` is in the first element slot

On that note, I'm a bit on the fence about the check-first-element-only approach, because it can seem unpredictable. An alternative would be to not try to resolve the ambiguity and just raise an error, similar to what we have been doing before this PR. But this requires users to always specify an Ibis schema manually for columns that contain arrays, even when their column data is clean enough for the type to be inferred accurately

Reply:
Let me know if you have any strong opinions (otherwise I can leave as-is, noting that this is something that is open to be reconsidered in the future)

Reply:
sure, note however that we shoul not be checking the first element at all! we should simply call `infer_dtype` on *each* element (if its not a scalar); pandas is designed to make this very performant and it will exit immediately for ndarrays with the dtype. 

e.g. something like

```
if is_scalar(e):
      if isna(e):
          continue # not sure what you need to track here
      raise? # (e.g. we don't allow scalars mixed i think)
else:
     inferred_type = infer_dtype(e)
     if ........
```

Reply:
Oh, I think this is promising. We _could_ check all elements of the Series—and it shouldn't be heavy because the only thing we need to do with each element really is to: 1) check if it's an array, and 2) call `dt.infer` on the array, which will either directly return the `np.array`'s dtype or rely on Pandas `infer_dtype` (also not heavy)

Reply:
Hmm the performance of checking all elements could end up being an issue: it looks like this would take ~5s for a Series with 1,000,000 arrays. It's hard for me to say whether this is reasonable or not. With many Series like this this would take a while, although maybe that wouldn't be common.

What do you think about leaving this as-is (I'm thinking that checking only the first element is not perfect but is an OK heuristic) and revisiting in a follow-up if necessary?